packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/studio/services/library/StudioLibraryPublicationHandler.java

```diff
@@ -0,0 +1,210 @@
+
+        if (optionalEditingContext.isPresent() && optionalEditingContext.get() instanceof EditingContext editingContext) {
+            ResourceSet rSet = editingContext.getDomain().getResourceSet();
+            rSet.getResourceFactoryRegistry().getProtocolToFactoryMap().put(IEMFEditingContext.RESOURCE_SCHEME, new JSONResourceFactory());
+
+            DependencyGraph<EObject> dependencyGraph = this.studioLibraryDependencyCollector.collectDependencies(rSet);
+
+            // Compute the topological ordering to ensure that all the dependencies of a library have been created before it.
+            List<EObject> libraryCandidates = dependencyGraph.computeTopologicalOrdering();
+
+            for (EObject libraryCandidate : libraryCandidates) {
+                if (libraryCandidate instanceof Domain || libraryCandidate instanceof RepresentationDescription) {
+                    Optional<String> optionalLibraryName = this.getUniqueLibraryName(libraryCandidate, libraryCandidates);
+                    if (optionalLibraryName.isPresent()) {
+                        Resource libraryResource = this.getOrCreateLibraryResource(input.projectId(), optionalLibraryName.get(), input.version(), rSet);
+                        libraryResource.getContents().add(libraryCandidate);
+                    }
+                } else {
+                    Resource sharedComponentsResource = this.getOrCreateLibraryResource(input.projectId(), "shared_components", input.version(), rSet);
+                    sharedComponentsResource.getContents().add(libraryCandidate);
+                }
+            }
+
+
+            Map<String, Library> createdLibraries = new HashMap<>();
+            for (EObject libraryCandidate : libraryCandidates) {
+                if (!createdLibraries.containsKey(this.getResourceName(libraryCandidate.eResource()))) {
+                    Optional<SemanticData> optionalSemanticData = this.toSemanticData(input, libraryCandidate.eResource());
+                    if (optionalSemanticData.isPresent()) {
+                        String libraryName = this.getResourceName(libraryCandidate.eResource());
+                        Library library = Library.newLibrary()
+                                .namespace(input.projectId())
+                                .name(libraryName)
+                                .semanticData(AggregateReference.to(optionalSemanticData.get().getId()))
+                                .dependencies(dependencyGraph.getDependencies(libraryCandidate).stream()
+                                        .map(dependency -> createdLibraries.get(this.getResourceName(dependency.eResource())).getId())
+                                        .distinct()
+                                        .map(AggregateReference::<Library, UUID>to)
+                                        .toList())
```

Comment:
It is not clear if the dependencies should be stored at the library level or the semantic data level. 
Storing it at the semantic data level would ease the process of loading editing context from libraries, since library dependencies would be treated as project dependencies.

Reply:
I suspect that this is right, I'll look into it.